### PR TITLE
Certificate Status Monitor Refactor

### PR DESCRIPTION
**Pull request by ArrisLee**
### Which issue this PR addresses:

https://issues.redhat.com/browse/ARO-13644

### What this PR does / why we need it:

- Resolve some logic bugs in the original code.
- Improve the readability
- Add more error handling and edge case checking

### Test plan for issue:

- Unit test
- CI/CD

### CODE REVIEW

**Comment by ArrisLee** [comment]
re-activate and rebase

**Comment by ArrisLee** [comment]
re-activate and rebase

**Comment by ArrisLee** [comment]
reactivate and rebase

**Review by kimorris27 — CHANGES_REQUESTED** [review]
It's a good refactor overall! The changes I'm suggesting are mostly regarding readability and potential for further refactoring.

File: pkg/monitor/cluster/certificateexpirationstatuses.go [hunk 1]
```diff
@@ -25,132 +25,178 @@ import (
...
 	}
 
+	// Retrieve and check the MDSD certificate via helper function.
+	if err := mon.processCertificate(ctx, operator.Namespace, operator.SecretName, genevalogging.GenevaCertName); err != nil {
+		// If the secret is not found, emit a missing secret metric.
+		if kerrors.IsNotFound(err) {
+			mon.emitGauge(secretMissingMetricName, 1, secretMissingMetric(operator.Namespace, operator.SecretName))
+			return nil
+		}
+		return fmt.Errorf("emitting MDSD certificate expiration metrics: %w", err)
```
[the remark this answers is not in the document]
Comment (ArrisLee) [reply pkg/monitor/cluster/certificateexpirationstatuses.go]: My first thought is to add more explanations to the error, but yeah, I agree with you, it doesn't add too much value on the current logic. I have modified these parts

File: pkg/monitor/cluster/certificateexpirationstatuses.go [hunk 2]
```diff
@@ -25,132 +25,178 @@ import (
...
-					"name":      secretName,
-					"namespace": operator.Namespace,
-				})
+		// Build a set of secret names to process, avoid duplicate processing.
+		secretNames := map[string]struct{}{
+			ingressSecretName: {},
+		}
+		// Also process the API Server certificate variant if it differs.
+		apiserverSecretName := strings.Replace(ingressSecretName, "-ingress", "-apiserver", 1)
+		secretNames[apiserverSecretName] = struct{}{}
```
[the remark this answers is not in the document]
Comment (ArrisLee) [reply pkg/monitor/cluster/certificateexpirationstatuses.go]: this part is untouched, I did not modify the logic. my understanding is, in some situations, the same certificate might be used in multiple places. For example, if the Ingress Controller and the API server share the same certificate, in that case, we do not need to process the cert twice.

**Review by kimorris27 — APPROVED** [review]
